pkg/operator/resource/resourceapply/generic.go

```diff
@@ -45,6 +45,10 @@ func init() {
 	utilruntime.Must(policyv1.AddToScheme(genericScheme))
 }
 
+// PreprocessResourcesFuncType gives the caller an opportunity to preprocess decoded resources
+// before applying them.
+type PreprocessResourcesFuncType func(ctx context.Context, object runtime.Object) (runtime.Object, error)
```

Comment:
naming suggestion: `PreprocessFunc`

Reply:
I'd like to leave it as is. It's symmetric with `PreprocessResourcesFuncType` in `static_resource_controller.go`.